### PR TITLE
Remove confusion between libsuitesparse the C library and SuiteSparse the stdlib

### LINKED ISSUE #40854: macOS: running `make` gives "WARNING: using mismatched version for..."
```
WARNING: using mismatched version for SuiteSparse-9cfdbbd33d2d52b83b2459f670162244cc95460c symlink-uninstaller /Users/dilum/dev/repos/JuliaLang/julia/usr/share/julia/stdlib/v1.7:
  want SuiteSparse.v5.8.1+2.x86_64-apple-darwin bb-uninstaller
  To resolve this warning, you could try either of the following suggestions:
  1. Run the following command: make -C deps uninstall
  2. Remove the following directory: /Users/dilum/dev/repos/JuliaLang/julia/usr
```

Per the discussion on Slack, the cause is a file uppercase/lowercase issue on macOS.

Specifically, on macOS, you see this (note that only `suitesparse` is present):
```bash
~/julia/usr/manifest$ ls
ArgTools       Statistics     dsfmt          libwhich       nghttp2        pcre
Downloads      Tar            gmp            llvm           objconv        suitesparse
LibCURL        blastrampoline libgit2        llvmunwind     openblas       utf8proc
NetworkOptions csl            libssh2        mbedtls        openlibm       zlib
Pkg            curl           libuv          mpfr           p7zip
```

But on a different system, you see this (note that both `suitesparse` and `SuiteSparse` are present):
```julia
~/julia/usr/manifest$ ls
ArgTools        curl       gmp      libssh2   llvm     NetworkOptions  openlibm  pcre        suitesparse  unwind
blastrampoline  Downloads  LibCURL  libuv     mbedtls  nghttp2         p7zip     Pkg         SuiteSparse  utf8proc
csl             dsfmt      libgit2  libwhich  mpfr     openblas        patchelf  Statistics  Tar          zlib
```

### PR DESCRIPTION
**Pull request by ViralBShah**
Fix #40854

### CODE REVIEW

File: deps/libsuitesparse.mk [hunk 1]
```diff
@@ -118,39 +118,39 @@ $(build_shlibdir)/libsuitesparse_wrapper.$(SHLIB_EXT): $(SRCDIR)/SuiteSparse_wra
...
+configure-libsuitesparse-wrapper:
+compile-libsuitesparse-wrapper:
+fastcheck-libsuitesparse-wrapper: #none
+check-libsuitesparse-wrapper:
+install-libsuitesparse-wrapper: $(build_shlibdir)/libsuitesparse_wrapper.$(SHLIB_EXT)
 
 else # USE_BINARYBUILDER_SUITESPARSE
 
-$(eval $(call bb-install,suitesparse,SUITESPARSE,false))
+$(eval $(call bb-install,libsuitesparse,SUITESPARSE,false))
```
Comment (staticfloat) [inline deps/libsuitesparse.mk:136]:
```suggestion
$(eval $(call bb-install,libsuitesparse,LIBSUITESPARSE,false))
```

**Review by staticfloat — CHANGES_REQUESTED** [review]
Good start; we'll also need to fix:

* https://github.com/JuliaLang/julia/blob/4ecbc16df572f54d3f05d872f13f31f9e6abc7ef/contrib/refresh_checksums.mk#L22
* https://github.com/JuliaLang/julia/blob/bb5b98e72a151c41471d8cc14cacb495d647fb7f/base/Makefile#L221-L228
* https://github.com/JuliaLang/julia/blob/bb5b98e72a151c41471d8cc14cacb495d647fb7f/Make.inc#L48 (and many other places that use `SUITESPARSE`)
* https://github.com/JuliaLang/julia/blob/a08a3ff1f957f6bd570ff5756dff4290eb6f43da/Makefile#L174

**Comment by ViralBShah** [comment]
How does it look now? The only place where it is now SUITESPARSE (instead of LIBSUITESPARSE) is in `stdlib/SuiteSparse.version`.

Can you see if `stdlib/Makefile` should use LIBSUITESPARSE? I wasn't quite sure.

**Comment by ViralBShah** [comment]
We really do sprinkle this stuff all over the place!

**Review by staticfloat — APPROVED** [review]

**Comment by ViralBShah** [comment]
Should we change the checksumming files to also use `libsuitesparse`?

**Comment by ViralBShah** [comment]
I thought these checksums are a problem with the SuiteSparse stdlib, and not the C library.

**Comment by KristofferC** [comment]
Yes, the last commit doesn't look right, those are indeed for the stdlib.

**Comment by ViralBShah** [comment]
I'm reverting the last commit.

**Comment by ViralBShah** [comment]
Is this possibly some kind of disk full thing?

```
/usr/local/bin/gtar: write error
Makefile:421: recipe for target 'binary-dist' failed
make: *** [binary-dist] Error 2
```

**Comment by KristofferC** [comment]
That feels likely to me.

**Comment by staticfloat** [comment]
Without the last commit, if you try to refresh the suitesparse checksums you don't get the proper packing; you get an empty `deps/checksums/libsuitesparse` file and a bunch of `deps/checksums/SuiteSparse*.tar.gz/` folders.  This is because there's now a name mismatch between the make variable LIBSUITESPARSE and the files on-disk.  My commit was an attempt to fix that, but clearly I messed something up.  I'll open a new PR.

**Comment by ViralBShah** [comment]
Ok. Sorry about that. I was mainly comfortable merging since it was green. Thanks for the new pr

**Pull request merged**